provider/aci_test.go

```diff
@@ -1191,3 +1191,247 @@ func TestCreatePodWithProjectedVolume(t *testing.T) {
+							Driver: "disk.csi.azure.com",
+							NodeStageSecretRef: &v1.SecretReference{
+								Name:      fakeVolumeSecret,
+								Namespace: podNamespace,
+							},
+							NodePublishSecretRef: &v1.SecretReference{
+								Name:      fakeVolumeSecret,
+								Namespace: podNamespace,
+							},
+						},
+					},
+				},
+			},
+			secretVolume: &fakeSecret,
+			volume: v1.Volume{
+				Name: pvName,
+				VolumeSource: v1.VolumeSource{
+					CSI: &v1.CSIVolumeSource{
+						Driver: "disk.csi.azure.com",
+						NodePublishSecretRef: &v1.LocalObjectReference{
+							Name: fakeVolumeSecret,
+						},
+						ReadOnly: &readOnly,
+					},
+				}},
+			expectedError: fmt.Errorf("Pod %s requires volume %s which is of an unsupported type %s", podName, pvName, "disk.csi.azure.com"),
+		},
+		{
+			description:      "Volume has NodePublishSecretRef with empty secret",
+			persistentVolume: &fakePV,
+			secretVolume:     nil,
+			volume: v1.Volume{
+				Name: pvName,
+				VolumeSource: v1.VolumeSource{
+					CSI: &v1.CSIVolumeSource{
+						Driver: "file.csi.azure.com",
+						NodePublishSecretRef: &v1.LocalObjectReference{
+							Name: "fakeEmptyVolumeSecret",
+						},
+						ReadOnly: &readOnly,
+					},
+				},
+			},
+			expectedError: fmt.Errorf("Getting secret for AzureFile CSI driver %s returned an empty secret", pvName),
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.description, func(t *testing.T) {
+
+			resourceManager, err := manager.NewResourceManager(mockPodLister, mockSecretLister, mockConfigMapLister, mockServiceLister, mockPvcLister, mockPvLister)
+			if err != nil {
+				t.Fatal("Unable to prepare the mocks for resourceManager", err)
+			}
+
+			provider, err := createTestProvider(aadServerMocker, aciServerMocker, resourceManager)
+			if err != nil {
+				t.Fatal("Unable to create test provider", err)
+			}
+
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      podName,
+					Namespace: podNamespace,
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name: "nginx",
+							ReadinessProbe: &v1.Probe{
+								Handler: v1.Handler{
+									HTTPGet: &v1.HTTPGetAction{
+										Port: intstr.FromInt(8080),
+										Path: "/",
+									},
+								},
+								InitialDelaySeconds: 10,
+								PeriodSeconds:       5,
+								TimeoutSeconds:      60,
+								SuccessThreshold:    3,
+								FailureThreshold:    5,
+							},
+						},
+					},
+				},
+			}
+
+			volumeMount := v1.VolumeMount{
+				Name:      pvName,
+				MountPath: "/temp",
+			}
+			pod.Spec.Containers[0].VolumeMounts = append(pod.Spec.Containers[0].VolumeMounts, volumeMount)
+
+			mockSecretLister.EXPECT().Secrets(podNamespace).Return(mockSecretNamespaceLister)
+
+			if tc.secretVolume == nil {
+				mockSecretNamespaceLister.EXPECT().Get(tc.volume.CSI.NodePublishSecretRef.Name).Return(nil, nil)
+			} else {
+				mockSecretNamespaceLister.EXPECT().Get(tc.secretVolume.Name).Return(tc.secretVolume, nil)
+			}
```

Comment:
I think this should be in the setup which mean the mockSecretNamespaceLister should be a parameter

Reply:
`mockSecretNamespaceLister` is not a parameter here, the return of the secret is the parameter (object or nil)